### PR TITLE
Changes related to etcd upgrade

### PR DESCRIPTION
**Pull request by bklei**
## Summary and Scope

Interim checkin of upgrade workflow, next steps revamp backup/restore -- and will then release chart(s) once that functionality is ready.

## Issues and Related PRs

* Resolves [CASMPET-6186](https://jira-pro.its.hpecorp.net:8443/browse/CASMPET-6186)

## Testing

Lots of upgrade testing in vshasta

### Tested on:

  * Virtual Shasta

### Test description:

Lots of upgrade testing in vshasta

## Risks and Mitigations

Low

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [ ] License file intact
- [ ] Target branch correct
- [ ] CHANGELOG.md updated
- [ ] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: charts/cray-etcd-base/templates/certificates.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {{/*
 MIT License
 
-(C) Copyright 2022 Hewlett Packard Enterprise Development LP
+(C) Copyright 2023 Hewlett Packard Enterprise Development LP
```
Comment (dborman-hpe) [inline charts/cray-etcd-base/templates/certificates.yaml:4]: Should be "2022-2023"

Comment (bklei) [reply charts/cray-etcd-base/templates/certificates.yaml:4]: good catch -- will fix

**Review by rgross727 — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -20,28 +20,33 @@ helm:
...
 
 package:
 ifdef CHART_VERSIONS
-	CMD="package charts/cray-etcd-backup   --version $(word 1, $(CHART_VERSIONS)) -d packages" $(MAKE) helm
-	CMD="package charts/cray-etcd-defrag   --version $(word 2, $(CHART_VERSIONS)) -d packages" $(MAKE) helm
-	CMD="package charts/cray-etcd-base --version $(word 3, $(CHART_VERSIONS)) -d packages" $(MAKE) helm
+	CMD="package charts/cray-etcd-backup          --version $(word 1, $(CHART_VERSIONS)) -d packages" $(MAKE) helm
+	CMD="package charts/cray-etcd-defrag          --version $(word 2, $(CHART_VERSIONS)) -d packages" $(MAKE) helm
+	CMD="package charts/cray-etcd-base            --version $(word 3, $(CHART_VERSIONS)) -d packages" $(MAKE) helm
+	CMD="package charts/cray-etcd-migration-setup --version $(word 3, $(CHART_VERSIONS)) -d packages" $(MAKE) helm
```
Comment (kimjensen-hpe) [inline Makefile:49]: Is 'word 3' for both cray-defrag and cray-etcd-migration-setup  correct?

File: charts/cray-etcd-base/templates/migration-jobs.yaml [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{ include "cray-etcd-base.fullname" . }}-etcd-post-install"
+  labels:
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+  annotations:
+    helm.sh/hook: post-upgrade, post-install
+    helm.sh/hook-delete-policy: before-hook-creation
+spec:
```
Comment (kimjensen-hpe) [inline charts/cray-etcd-base/templates/migration-jobs.yaml:131]: do you want a ttlSecondsAfterFinished on this job as well?

**Comment by mtupitsyn** [comment]
The `charts/cray-etcd-base/charts/etcd-8.5.10.tgz` is probably committed by mistake...
Also, I'd recommend to add `requirements.lock` to `.gitignore` - this file is auto-generated and may change unpredictably.

**Review by kimjensen-hpe — APPROVED** [review]

**Review by bo-quan — APPROVED** [review]

File: charts/cray-etcd-base/templates/migration-jobs.yaml (hunk 3 above)
Comment (bklei) [reply charts/cray-etcd-base/templates/migration-jobs.yaml:131]: great catch -- yes i do!  will fix..

File: Makefile (hunk 2 above)
Comment (bklei) [reply Makefile:49]: fixed

Comment (bklei) [reply Makefile]: fixed -- not sure how it was working, maybe I would have had a publish failure.  great catch.

**Pull request merged**